### PR TITLE
Added merch products page

### PR DESCRIPTION
**Pull request by limivann**
This PR adds a product page with dummy data. 

Ticket: https://github.com/orgs/ntuscse/projects/3?pane=issue&itemId=49452950

### CODE REVIEW

**Review by Iyzyman — DISMISSED** [review]

**Review by Iyzyman — DISMISSED** [review]

**Review by Iyzyman — APPROVED** [review]

**Review by Iyzyman — APPROVED** [review]

File: apps/cms/src/@types/Product.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  constructor(
+    public id: string = "",
+    public name: string = "",
+    public price: number = 0, // Price of the product in cents ($19 = 1900).
+    public images: string[] = [], // URL of the images.
+    public sizes: ProductSizeTypes[] = [],
+    public productCategory: ProductCategoryType = "",
+    public isAvailable?: boolean
+  ) {}
+}
```
Comment (chanbakjsd) [inline apps/cms/src/@types/Product.ts:11]: Is it necessary to make this a `class` (will there be methods)?
Would it make more sense to declare it as a type instead?

File: apps/cms/src/admin/utils/RenderCellFactory.tsx [hunk 2]
```diff
@@ -2,42 +2,105 @@ import React from "react";
 import payload from "payload";
 
 export class RenderCellFactory {
-
   static get(element: unknown, key: string) {
-    console.log(key)
+    console.log("key", key);
```
Comment (chanbakjsd) [inline apps/cms/src/admin/utils/RenderCellFactory.tsx:6]: Consider removing the debug statement.

File: apps/cms/src/@types/Product.ts (hunk 1 above)
Comment (Iyzyman) [reply apps/cms/src/@types/Product.ts:11]: Hey Wen Xu, for this class i think Ivan has created it mainly just to create some mock data for the products front end page. The proper products types and api will be implemented by the backend team in the coming week. So for the time being is ok to commit with this mock version?

Comment (Iyzyman) [reply apps/cms/src/@types/Product.ts:11]: > Is it necessary to make this a `class` (will there be methods)? Would it make more sense to declare it as a type instead?

Hey Wen Xu, for this class i think Ivan has created it mainly just to create some mock data for the products front end page. The proper products types and api will be implemented by the backend team in the coming week. So for the time being is ok to commit with this mock version?

Comment (limivann) [reply apps/cms/src/@types/Product.ts:11]: @chanbakjsd It must be a class for keys in payload columns.

**Pull request closed without merging**